### DISCUSSION TITLE
Cant access redux store inside getInitialProps

### DISCUSSION THREAD

**Discussion by ipirm** [vercel/next.js#14873] [Help]
Store is undefined 
    Post.getInitialProps({store}) {
    console.log(store)
        return {}
    }

**Comment by jamesmosier** [comment]
Do you have any more code you could share?

**Reply by ipirm** [reply]
import Router, {useRouter} from 'next/router'
import {Layout} from "../../components/layout/Layout";
import Skeleton from "react-loading-skeleton";
import React, {useEffect} from "react";
import { getCompany} from "../../store/actions/post";
import {useDispatch, useSelector} from 'react-redux'
import PostSwiper from "../../components/elements/PostSwiper";
import PostSwiperScelet from "../../components/sceletons/PostSwiperScelet";
import MapGoogle from "../../components/elements/MapGoogle";

const Post = () => {
    const router = useRouter()
    const {pid} = router.query
    const dispatch = useDispatch()

    useEffect(() => {
        dispatch(getCompany(pid))
    }, [dispatch, pid]);

    const loading = useSelector(state => state.post.loading)

    const company = useSelector(state => state.post.company)



    return (
        <Layout>
            <div className="company-section company-page">
                <div className="container-fluid swiper-fluid">
                    <div className="row" style={{justifyContent: 'space-between', alignItems: 'center'}}>
                        <div className="col-auto">
                            <a className="post-back-button" onClick={() => Router.back()}>
                                <svg width="26" height="30" viewBox="0 0 26 30" fill="none"
                                     xmlns="http://www.w3.org/2000/svg">
                                    <g clipPath="url(#clip0)">
                                        <path
                                            d="M18.1212 10.884H8.57944V6.31567C8.57512 5.93723 8.30579 5.63451 7.9778 5.6395C7.81517 5.64194 7.66058 5.72125 7.55001 5.85888L0.185813 14.5387C-0.0328183 14.7483 -0.062828 15.1227 0.118746 15.375C0.13898 15.403 0.161411 15.429 0.185813 15.4524L7.55001 24.1322C7.66376 24.271 7.82002 24.3531 7.98553 24.3606H8.22311C8.44917 24.234 8.58898 23.9651 8.57944 23.6753V19.5638H18.1212C18.5199 19.4858 18.8263 19.1137 18.8734 18.6502V11.3408C18.8734 10.9754 18.4775 10.884 18.1212 10.884Z"
                                            fill="#F8C563"/>
                                        <path
                                            d="M20.6554 10.6555C20.3274 10.6555 20.0615 10.9624 20.0615 11.3408V18.6502C20.0615 19.0286 20.3274 19.3355 20.6554 19.3355C20.9834 19.3355 21.2493 19.0286 21.2493 18.6502V11.3408C21.2493 10.9624 20.9835 10.6555 20.6554 10.6555Z"
                                            fill="#F8C563"/>
                                        <path
                                            d="M23.0304 10.6555C22.7024 10.6555 22.4365 10.9624 22.4365 11.3408V18.6502C22.4365 19.0286 22.7024 19.3355 23.0304 19.3355C23.3584 19.3355 23.6243 19.0286 23.6243 18.6502V11.3408C23.6243 10.9624 23.3584 10.6555 23.0304 10.6555Z"
                                            fill="#F8C563"/>
                                        <path
                                            d="M25.4064 10.6555C25.0784 10.6555 24.8125 10.9624 24.8125 11.3408V18.6502C24.8125 19.0286 25.0784 19.3355 25.4064 19.3355C25.7344 19.3355 26.0003 19.0286 26.0003 18.6502V11.3408C26.0003 10.9624 25.7344 10.6555 25.4064 10.6555Z"
                                            fill="#F8C563"/>
                                    </g>
                                    <defs>
                                        <clipPath id="clip0">
                                            <rect width="26" height="30" fill="white"/>
                                        </clipPath>
                                    </defs>
                                </svg>
                                <span>Назад</span>
                            </a>
                        </div>
                    </div>
                    <div className="container post-container">
                        <div className="row">
                            <div className="col-12">
                                <div className="d-flex">
                                    <div className="post-text">
                                        <div className="post-title"><span>{loading ? '' : company.name}</span></div>
                                        <div className="post-description">
                                            {company.text}
                                        </div>
                                        <div className="post-information">
                                            <p>Адреса магазинов:</p>
                                            <span>{company.address}</span>
                                            <div className="post-information-time">Часы работы: {company.time}</div>
                                        </div>
                                        <div className="post-information">
                                            <p>Телефоны:</p>
                                            <div className="post-information-time">{company.number}</div>
                                        </div>
                                    </div>
                                        <img className="post-image" src={company ? company.image_url : ''}/>
                                </div>
                            </div>
                        </div>
                    </div>
                    {/*<MapGoogle />*/}
                    {/*{*/}
                    {/*    postsSame !== null && !loading*/}
                    {/*        ?*/}
                    {/*        <PostSwiper companies={companies} cats={cats} items={postsSame} label="Похожие Компании"/>*/}
                    {/*        :*/}
                    {/*        <PostSwiperScelet items={Array(9).fill()} />*/}
                    {/*}*/}
                </div>
            </div>
        </Layout>
    )
}
Post.getInitialProps = async ({ query, store}) => {
    const res = await fetch(`https://www.covid.info.az/api/posts-company/${query.pid}`)
    console.log(store) // i am getting undefined here
    const post = await res.json()
    return {post}
}
export default Post

**Reply by ipirm** [reply]
import React from 'react'
import {Provider} from 'react-redux'
import {createWrapper} from 'next-redux-wrapper'
import store from '../store'
import {AnimatePresence} from 'framer-motion'
import NProgress from 'nprogress'
import 'nprogress/nprogress.css'
import Router from 'next/router'


Router.events.on('routeChangeStart', () => NProgress.start())
Router.events.on('routeChangeComplete', () => NProgress.done())
Router.events.on('routeChangeError', () => NProgress.done())


const  MyApp = ({Component, pageProps,router}) => {
    return(
            <AnimatePresence exitBeforeEnter>
                <Provider store={store}>
                    <Component {...pageProps} key={router.route}/>
                </Provider>
            </AnimatePresence>
    )
}
const makeStore = () => store;
const wrapper = createWrapper(makeStore)

export default wrapper.withRedux(MyApp)

**Reply by ipirm** [reply]
import {createStore, combineReducers, applyMiddleware} from 'redux'
import {postReducer} from './reducers/post'
import ReduxThunk from 'redux-thunk'

const rootReducer = combineReducers({
    post: postReducer
})

export default createStore(rootReducer, applyMiddleware(ReduxThunk))

**Reply by ipirm** [reply]
const  MyApp = ({Component, pageProps,router}) => {
    return(
        AnimatePresence exitBeforeEnter>
                Provider store={store}>
                    <Component {...pageProps} key={router.route}/>
                /Provider>
            /AnimatePresence>
    )
}
